tool/tsh/common/logger.go

```diff
@@ -52,6 +60,25 @@ func initLogger(cf *CLIConf, opts loggingOpts) error {
 	return trace.Wrap(utils.InitLogger(utils.LoggingForCLI, level, initLoggerOpts...))
 }
 
+// initMCPLogger initializes a logger to be used on MCP servers.
+func initMCPLogger(cf *CLIConf) (*slog.Logger, error) {
+	opts := parseLoggingOptsFromEnvAndArgv(cf)
+	cf.OSLog = opts.osLog
+	cf.Debug = opts.debug || opts.osLog
+
+	level := slog.LevelInfo
+	if cf.Debug {
+		level = slog.LevelDebug
+	}
+
+	logger, _, err := logutils.Initialize(logutils.Config{
+		Severity: level.String(),
+		Format:   mcpLogFormat,
+		Output:   mcpLogOutput,
+	})
+	return logger, trace.Wrap(err)
```

Comment:
does this conflict with os log?

Reply:
Could you clarify what you mean?

Reply:
What happens when `--os-log` flag is true?
https://goteleport.com/docs/ver/18.x/connect-your-client/tsh/#unified-logging-system-on-macos

 `--os-log` should have higher priority in my opinion. or we can disallow `--os-log` for mcp, or log both. we shouldn't ignore the flag though.

Reply:
I've updated the logger part to share the same flow as regular `tsh`. I only added a new MCP logger purpose to set the format and output appropriately.

Reply:
```
	switch {
	case o.osLogSubsystem != "":
		output = logutils.LogOutputOSLog
	case purpose == LoggingForMCP:
		output = logutils.LogOutputMCP
		o.format = LogFormatJSON
```
I like the purpose for MCP. does this logic mean `o.osLogSubsystem` has higher priority? could we leave a comment on that? also do really need `utils.InitLogger` to return the logger?

Reply:
> does this logic mean o.osLogSubsystem has higher priority?

It should have, I'll add a comment on that.

> also do really need utils.InitLogger to return the logger?

It is not necessary, but I feel it is better to have it passed around instead of relying in the global instance. I can update that to use the global instance if it makes sense.